### PR TITLE
Cosmos Encryption removed block() in implementation

### PR DESCRIPTION
**Pull request by moderakh**
* This PR has 0 change in azure-cosmos
* This PR scope is to remove block() calls in cosmos-encryption lib implementation.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/AzureKeyVaultKeyWrapProvider.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public AzureKeyVaultKeyWrapProvider(KeyVaultTokenCredentialFactory keyVaultToken
 
     // TODO make this async
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/AzureKeyVaultKeyWrapProvider.java:51]: NIT: Remove TODO comment?

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/CachedDekProperties.java [hunk 2]
```diff
@@ -13,7 +13,7 @@ public CachedDekProperties(
         DataEncryptionKeyProperties serverProperties,
         Instant serverPropertiesExpiryUtc) {
         assert(serverProperties != null);
-
+        // TODO: add NPE validation
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/CachedDekProperties.java:16]: NIT: wouldn't it make sense to do that in this PR vs. just adding TODO? Or is the work more involved than I naively assume?

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/DataEncryptionKeyContainerCore.java [hunk 3]
```diff
@@ -96,39 +99,42 @@ public DataEncryptionKeyContainerCore(CosmosDataEncryptionKeyProvider dekProvide
...
+
+                return wrapResultMono.flatMap(
+                    wrapResult -> {
+                        byte[] wrappedDek = wrapResult.getLeft();
+                        EncryptionKeyWrapMetadata updatedMetadata = wrapResult.getMiddle();
+                        InMemoryRawDek updatedRawDek = wrapResult.getRight();
+
+                        CosmosItemRequestOptions effectiveRequestOptions = requestOptions != null ? requestOptions : new CosmosItemRequestOptions();
+
+                        effectiveRequestOptions.setIfMatchETag(dekProperties.eTag);
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/DataEncryptionKeyContainerCore.java:116]: You are modifying the passed in request options here. Wouldn't it be better to create a deep copy instead?

**Review by FabianMeiswinkel — APPROVED** [review]
Couple of NITs - but otherwise LGTM

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/CachedDekProperties.java (hunk 2 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/CachedDekProperties.java:16]: TODO comment is not accurate. will update the TODO to capture the details. 

on contentResponseOnWriteEnabled = false we get into NPE.
I need to think on what to do on that case (for contentResponseOnWriteEnabled =false). 
Just NPE validation won't help here.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/AzureKeyVaultKeyWrapProvider.java (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/AzureKeyVaultKeyWrapProvider.java:51]: thanks will do.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/CachedDekProperties.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/CachedDekProperties.java:16]: Makes sense

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/AzureKeyVaultKeyWrapProvider.java (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/AzureKeyVaultKeyWrapProvider.java]: done.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/DataEncryptionKeyContainerCore.java (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/DataEncryptionKeyContainerCore.java:118]: addressed. good point. thank you.

**Pull request merged**